__test__/unit/adapters/collateralized_simple_interest_loan_adapter.spec.ts

```diff
@@ -239,3 +254,364 @@ describe("Collateralized Terms Contract Interface (Unit Tests)", () => {
+                ...debtRegistryEntryBase,
+                termsContractParameters: debtOrderForScenario3.termsContractParameters,
+            },
+        };
+    });
+
+    describe("#toDebtOrder", () => {
+        describe("collateralized simple interest loan's required parameter is missing or malformed", () => {
+            describe("`collateralTokenSymbol` is missing", () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.toDebtOrder({
+                            ...scenario_1.minimalLoanOrder,
+                            collateralTokenSymbol: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "collateralTokenSymbol"');
+                });
+            });
+            describe("`collateralTokenSymbol` is not tracked by Token Registry", () => {
+                test("should throw CANNOT_FIND_TOKEN_WITH_SYMBOL", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.toDebtOrder({
+                            ...scenario_1.minimalLoanOrder,
+                            collateralTokenSymbol: "EOS", // EOS is not tracked in our test env's registry
+                        }),
+                    ).rejects.toThrow(ContractsError.CANNOT_FIND_TOKEN_WITH_SYMBOL("EOS"));
+                });
+            });
+            describe("`collateralAmount` is missing", async () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.toDebtOrder({
+                            ...scenario_1.minimalLoanOrder,
+                            collateralAmount: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "collateralAmount"');
+                });
+            });
+            describe("`gracePeriodInDays` is missing", async () => {
+                test("should throw DOES_NOT_CONFORM_TO_SCHEMA", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.toDebtOrder({
+                            ...scenario_1.minimalLoanOrder,
+                            gracePeriodInDays: undefined,
+                        }),
+                    ).rejects.toThrow('instance requires property "gracePeriodInDays"');
+                });
+            });
+        });
+
+        describe("collateralized simple interest loan's required parameters are present and well-formed ", () => {
+            describe("Scenario #1", () => {
+                test("should return debt order with correctly packed values", async () => {
+                    await expect(
+                        collateralizedSimpleInterestLoanAdapter.toDebtOrder(
+                            scenario_1.fullLoanOrder,
```

Comment:
Shouldn't this be the `minimalLoanOrder`?  The whole point here of having flexibility in terms of what object params need to be passed into the adapter is to make sure that users don't have to suss out what the Kernel's current address is and pass it in as a "full loan order" param in order to generate a debt order.

Reply:
Happy to use `minimalLoanOrder` in these cases but the combination of types and method signatures should be enforcing the desired behavior you outlined.